core/src/ephemeral_server/mod.rs

```diff
@@ -0,0 +1,461 @@
+use anyhow::anyhow;
+use flate2::read::GzDecoder;
+use futures::StreamExt;
+use serde::Deserialize;
+#[cfg(target_family = "unix")]
+use std::os::unix::fs::OpenOptionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    path::{Path, PathBuf},
+};
+use temporal_client::ClientOptionsBuilder;
+use tokio::{
+    task::spawn_blocking,
+    time::{sleep, Duration},
+};
+use tokio_util::io::{StreamReader, SyncIoBridge};
+use url::Url;
+use zip::read::read_zipfile_from_stream;
+
+/// Configuration for Temporalite.
+#[derive(Debug, Clone, derive_builder::Builder)]
+pub struct TemporaliteConfig {
+    /// Required path to executable or download info.
+    pub exe: EphemeralExe,
+    /// Namespace to use.
+    #[builder(default = "\"default\".to_owned()")]
+    pub namespace: String,
+    /// IP to bind to.
+    #[builder(default = "\"127.0.0.1\".to_owned()")]
+    pub ip: String,
+    /// Port to use or obtains a free one if none given.
+    #[builder(default)]
+    pub port: Option<u16>,
+    /// Sqlite DB filename if persisting or non-persistent if none.
+    #[builder(default)]
+    pub db_filename: Option<String>,
+    /// Whether to enable the UI.
+    #[builder(default)]
+    pub ui: bool,
+    /// Log format and level
+    #[builder(default = "(\"pretty\".to_owned(), \"warn\".to_owned())")]
+    pub log: (String, String),
```

Comment:
nit: Could have a separate struct for log instead of this tuple.

Reply:
There wasn't much value to expose a whole new module-level type for this.